### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by merrilmathew**
This PR includes github actions to deploy monitoring solution to prod environment. Added variable files and updated Makefile to include prod target.

### CODE REVIEW

File: terraform/workspace_variables/preprod.backend.tfvars [hunk 1]
```diff
@@ -0,0 +1,3 @@
+storage_account_name = "s165t01preprodtfstate"
+key                  = "preprod_monitoring.tfstate"
+resource_group_name  = "s165t01-preprod-rg"
```
Comment (chelnak) [inline terraform/workspace_variables/preprod.backend.tfvars]: Missing a new line

File: terraform/workspace_variables/preprod.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "key_vault_secret_name": "MONITORING",
+    "paas_space": "tra-test",
+    "monitoring_instance_name": "tra-monitoring-preprod",
+    "monitor_scrape_applications": [
+      "qualified-teachers-api-preprod.london.cloudapps.digital"
+    ],
+    "monitor_scrape_backing_services": [
+      "tra-test/qualified-teachers-api-preprod-pg-svc"
+    ]
+}
```
Comment (chelnak) [inline terraform/workspace_variables/preprod.tfvars.json]: Missing a new line

File: terraform/workspace_variables/prod.backend.tfvars [hunk 3]
```diff
@@ -0,0 +1,3 @@
+storage_account_name = "s165p01tfstate"
+key                  = "prod_monitoring.tfstate"
+resource_group_name  = "s165p01-rg"
```
Comment (chelnak) [inline terraform/workspace_variables/prod.backend.tfvars]: Missing a new line

File: terraform/workspace_variables/prod.tfvars.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "key_vault_secret_name": "MONITORING",
+    "paas_space": "tra-production",
+    "monitoring_instance_name": "tra-monitoring-prod",
+    "monitor_scrape_applications": [
+      "qualified-teachers-api-prod.london.cloudapps.digital"
+    ],
+    "monitor_scrape_backing_services": [
+      "tra-production/qualified-teachers-api-prod-pg-svc"
+    ]
+}
```
Comment (chelnak) [inline terraform/workspace_variables/prod.tfvars.json]: Missing a new line

File: terraform/workspace_variables/test.tfvars.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    "key_vault_secret_name": "MONITORING",
+    "paas_space": "tra-test",
+    "monitoring_instance_name": "tra-monitoring-test",
+    "monitor_scrape_applications": [
+      "qualified-teachers-api-test.london.cloudapps.digital"
+    ],
+    "monitor_scrape_backing_services": [
+      "tra-test/qualified-teachers-api-test-pg-svc"
+    ]
+  }
```
Comment (chelnak) [inline terraform/workspace_variables/test.tfvars.json]: Missing a new line

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -28,9 +29,94 @@ jobs:
         terraform init -backend=false
         terraform validate -no-color
       working-directory: terraform
-
+
```
Comment (chelnak) [inline .github/workflows/build.yml]: is this whitespace char needed?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -28,9 +29,94 @@ jobs:
...
+      with:
+        environment_name: preprod
+        azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+  deploy_prod:
+    name: Deploy to prod environment
+    needs: deploy_preprod
+    runs-on: ubuntu-latest
+    environment: prod
+    # if: github.ref == 'refs/heads/main'
```
Comment (chelnak) [inline .github/workflows/build.yml]: Is this redundant now?

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -28,9 +29,94 @@ jobs:
...
+      run: |
+        terraform init -backend=false
+        terraform validate -no-color
+      working-directory: terraform
+
+    - name: Deploy to prod environment
+      uses: ./.github/workflows/actions
+      with:
+        environment_name: prod
+        azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
```
Comment (chelnak) [inline .github/workflows/build.yml]: Missing a new line

**Review by chelnak — CHANGES_REQUESTED** [review]
Some minor changes requested & have asked for clearer commit messages

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -28,9 +29,94 @@ jobs:
...
+    runs-on: ubuntu-latest
+    environment: test
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 1.0.10
+
+    - name: Check formatting
```
Comment (saliceti) [inline .github/workflows/build.yml]: Formatting and validation were already done in dev, we probably don't need to do it again

File: .github/workflows/build.yml (hunk 8 above)
Comment (saliceti) [reply .github/workflows/build.yml]: @merrilmathew you may want to enable these options in VScode:
- Files: Insert final newline
- Files: Trim final newlines

File: .github/workflows/build.yml (hunk 6 above)
Comment (saliceti) [reply .github/workflows/build.yml]: @merrilmathew you may want to enable this option in VScode:
- Trim trailing whitespace

**Comment by merrilmathew** [comment]
Think I resolved the suggestions.

File: .github/workflows/build.yml (hunk 7 above)
Comment (merrilmathew) [reply .github/workflows/build.yml]: Removed. There is environment check instead

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
     - dev
+    - main
```
Comment (saliceti) [inline .github/workflows/build.yml:6]: Can we leave `main` only?

File: .github/workflows/destroy/action.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
+name: Destroy PAAS environment
```
Comment (saliceti) [inline .github/workflows/destroy/action.yml]: As discussed, we can remove this action and use the make commands instead if necessary

File: .github/workflows/build.yml (hunk 10 above)
Comment (merrilmathew) [reply .github/workflows/build.yml:6]: removed dev

File: .github/workflows/destroy/action.yml (hunk 11 above)
Comment (merrilmathew) [reply .github/workflows/destroy/action.yml]: removed action and rules from Makefile

**Comment by chelnak** [comment]
10/10 would read the commits again 👍

**Review by chelnak — APPROVED** [review]

**Pull request merged**